client-misk-dynamodb/src/test/kotlin/app/cash/backfila/client/misk/dynamodb/DynamoDbLastEvaluatedKeyTest.kt

```diff
@@ -0,0 +1,79 @@
+package app.cash.backfila.client.misk.dynamodb
+
+import app.cash.backfila.client.misk.BackfillConfig
+import app.cash.backfila.client.misk.TestingModule
+import app.cash.backfila.client.misk.embedded.Backfila
+import app.cash.backfila.client.misk.embedded.createWetRun
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper
+import javax.inject.Inject
+import misk.aws.dynamodb.testing.DockerDynamoDb
+import misk.testing.MiskExternalDependency
+import misk.testing.MiskTest
+import misk.testing.MiskTestModule
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+
+@MiskTest(startService = true)
+class DynamoDbLastEvaluatedKeyTest {
+  @Suppress("unused")
+  @MiskTestModule
+  val module = TestingModule()
+
+  @Suppress("unused")
+  @MiskExternalDependency
+  val dockerDynamoDb = DockerDynamoDb
+
+  @Inject lateinit var backfila: Backfila
+  @Inject lateinit var testData: DynamoMusicTableTestData
+
+  @Test
+  fun `one big segment with one second batch execution time works`() {
+    testData.addThriller()
+
+    // TODO(mikepaw) use batch size to change how many items show up in a batch.
+    val run = backfila.createWetRun<PausingBackfill>(
+        parameters = PausingBackfill.Parameters(1, 1, 128L)
+    )
+    run.execute()
+
+    val rows = testData.getTracksDump()
+    assertThat(rows).extracting<String> { it.track_title }.allMatch { it.endsWith("(EXPLICIT)") }
+  }
+
+  @Test
+  fun `really long run batches work`() {
+    testData.addLinkinPark()
+
+    // TODO(mikepaw) use batch size to change how many items show up in a batch.
```

Comment:
I need the batch size to be passed down with the RunBatch for me to be more accurate here. Currently it is set to a default of 4. This will be my next PR.